crates/miden-batch-prover/Cargo.toml

```diff
@@ -0,0 +1,48 @@
+[package]
+name = "miden-batch-prover"
+version = "0.8.0"
+description = "Miden rollup batch executor and prover"
+readme = "README.md"
+categories = ["no-std"]
+keywords = ["miden", "batch", "prover"]
+license.workspace = true
+authors.workspace = true
+homepage.workspace = true
+repository.workspace = true
+rust-version.workspace = true
+edition.workspace = true
+
+[lib]
+bench = false
+
+[features]
+default = ["std"]
+std = [
+  "assembly/std",
+  "miden-objects/std",
+  "miden-tx/std",
+  "miden-crypto/std",
+  "miden-verifier/std",
+  "vm-core/std",
+  "vm-processor/std",
+]
+# TODO: Check if we need this feature.
+testing = ["dep:rand", "dep:miden-lib", "dep:winterfell", "dep:anyhow"]
+
+[dependencies]
+anyhow = { version = "1.0", default-features = false, optional = true }
+assembly = { workspace = true }
+miden-crypto = { workspace = true }
+miden-lib = { workspace = true, optional = true, features = ["testing"] }
+miden-tx = { workspace = true }
+miden-objects = { workspace = true }
+miden-verifier = { workspace = true }
+rand = { workspace = true, optional = true, features = ["small_rng"] }
+thiserror = { workspace = true }
+vm-core = { workspace = true }
+vm-processor = { workspace = true }
+winterfell = { version = "0.11", optional = true }
```

Comment:
Could we double-check if all of these are needed as dependencies or if some could go into dev-dependencies? (e.g., upon a quick look I didn't see if we are using `rand` in non-testing code).